src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -50,6 +50,15 @@
 
     private static final String ERROR_NOT_INSTALLED = ERROR_AUTHENTICATING_GITHUB_APP + NOT_INSTALLED;
 
+    /**
+     * When a new {@link AppInstallationToken} is generated, wait this many seconds before continuing.
+     * Has no effect when a cached token is used, only when a new token is generated.
+     *
+     * Provided as one more possible avenue for debugging/stabilizing JENKINS-62249.
+     */
+    private static long AFTER_TOKEN_GENERATION_DELAY_SECONDS =
+        Long.getLong(GitHubAppCredentials.class.getName() + ".AFTER_TOKEN_GENERATION_DELAY_SECONDS", 0);
```

Comment:
I think you possibly meant:
```suggestion
        SystemProperties.getLong(GitHubAppCredentials.class.getName() + ".AFTER_TOKEN_GENERATION_DELAY_SECONDS", 0);
```

not that it requires a minimum core version of 2.236 to use that API unless you suppress the enforcer (which could be legitimate as it is a public API with no changes later on)

or

```suggestion
        Long.getLong(System.getProperty(GitHubAppCredentials.class.getName() + ".AFTER_TOKEN_GENERATION_DELAY_SECONDS", "0"));
```

Reply:
I think the existing code is right. [`Long.getLong(String)`](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#getLong-java.lang.String-) looks up the system property using the name you pass in, and [`Long.getLong(String, Long)`](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#getLong-java.lang.String-java.lang.Long-) lets you supply a default value. Same for `Integer.getInteger`, `Boolean.getBoolean`, etc. The first suggestion would be equivalent to the existing code, but the second proposal would look up the system property twice.

Reply:
> note that it requires a minimum core version of 2.236 to use that API unless you suppress the enforcer (which could be legitimate as it is a public API with no changes later on)

I'm going to leave it then.  I will use that facility when it doesn't involve turning off checks.